src/sinks/util/http.rs

```diff
@@ -127,12 +133,14 @@ where
     }
 }
 
-impl<T, B, L> Sink<Event> for BatchedHttpSink<T, B, L>
+impl<T, B, R, L> Sink<Event> for BatchedHttpSink<T, B, R, L>
 where
     B: Batch,
     B::Output: Clone + Send + 'static,
     T: HttpSink<Input = B::Input, Output = B::Output>,
-    L: RetryLogic<Response = http::Response<Bytes>> + Send + 'static,
+    R: From<usize> + fmt::Debug + Send + 'static,
+    L: RetryLogic<Response = (hyper::Response<Bytes>, R)> + Send + 'static,
+    (hyper::Response<Bytes>, R): sink::Response,
```

Comment:
Would this benefit from a new type wrapper or trait instead of just being a tuple here?

Reply:
While I am happy to add a wrapper, I do not think that this will change a lot. Instead of `(hyper::Response<Bytes>, R)` we will need something `ServiceResponse<R>`.